File: scala-example/spring-boot-example/src/main/scala/com/github/poad/scala/springboot/example/repository/MessageRepository.scala
package com.github.poad.scala.springboot.example.repository

import com.github.poad.scala.springboot.example.entity.MessageEntity
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.{CrudRepository, RepositoryDefinition}

@RepositoryDefinition(domainClass = classOf[MessageEntity], idClass = classOf[String])
trait MessageRepository extends CrudRepository[MessageEntity, String] {

  @Query(value = "SELECT UUID();", nativeQuery = true)
  def uuid(): String
}

File: java-example/spring-boot/spring-boot-jpa-example/src/main/java/com/github/poad/example/springboot/jpa/repository/MessageRepository.java
package com.github.poad.example.springboot.jpa.repository;

import com.github.poad.example.springboot.jpa.entity.MessageEntity;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.RepositoryDefinition;

/**
 * Created by ken-yo on 2016/08/06.
 */
@RepositoryDefinition(domainClass = MessageEntity.class, idClass = String.class)
public interface MessageRepository extends CrudRepository<MessageEntity, String> {
}
